### ISSUE TITLE
huge speed increase

### ISSUE THREAD

**Issue by jondot** [rust-analyzer/rust-analyzer#3188]
Wow! thanks team. Just updated after 2-3 weeks of not updating, I'm getting head-spinning speed increase ❤️ 

Feel free to close -- this is just feedback and to say thanks!

**Comment by matklad** [comment]
Thanks <3 !

I guess we need to close this once we regress performance back :D

**Comment by bjorn3** [comment]
Should I add `std::thread::sleep_ms(10_000)` to the main loop to fix this? :)

**Comment by matklad** [comment]
That's too easy: I suggest recompiling all the code with cranelift on every keystroke :-)

**Comment by bjorn3** [comment]
That's even better! That would increase the quality of my code, because it will be tested on more projects.

Another idea would be to compile rust-analyzer itself using the tls branch of cg_clif. (I should actually try that)

**Comment by matklad** [comment]
Right, the current workaround is to compile rust-analyzer in debug mode :)

Also, I'll unsubscribe myself from this issue, otherwise I'll continue making silly jokes indefinitely (which probably will actually help with not making rust-analyzer slower...)

**Comment by jondot** [comment]
🤣

**Comment by StackDoubleFlow** [comment]
This is creating real issues with my procrastination workflow. pls fix

**Comment by jondot** [comment]
Hey all,
Just wanted to bump this -- and say I've just experienced a huge speed bump again.
I'm working on the same project as in the last 1.5 years, and now I'm getting code completions as fast as Golang code completion (which is crazy!).
I'm now completely happy with the experience!
